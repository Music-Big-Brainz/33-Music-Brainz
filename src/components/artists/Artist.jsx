import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const Artist = ({ artistId, name, disambiguation }) => (
  <Link to={`/releases/${artistId}`}>
    <figure>
      <p>{name}</p>
      <p>{disambiguation}</p>
    </figure>
  </Link>
);

Artist.propTypes = {
  artistId: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  disambiguation: PropTypes.string.isRequired,
};

export default Artist;
