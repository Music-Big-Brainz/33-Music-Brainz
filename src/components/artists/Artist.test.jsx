import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Artist from './Artist';

describe('displays an Artist', () => {
  afterEach(() => cleanup());
  it('renders one single artist', () => {
    const { asFragment } = render(<Artist artistId="" name="" disambiguation />);
    expect(asFragment()).toMatchSnapshot();
  });
});
