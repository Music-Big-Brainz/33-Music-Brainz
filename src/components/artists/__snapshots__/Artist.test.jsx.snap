// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays an Artist renders one single artist 1`] = `
<DocumentFragment>
  <a
    href="/releases//"
  >
    <figure>
      <p />
      <p />
    </figure>
  </a>
</DocumentFragment>
`;
